apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrBuilder.kt

```diff
@@ -132,8 +133,8 @@ internal class IrBuilder(
       visitedTypes.add(name)
       val typeDefinition = schema.typeDefinition(name)
       if (typeDefinition.isBuiltIn()) {
-        // We don't generate builtin types
-        continue
+        // We don't generate builtin types, unless they are explicitly mapped
```

Comment:
```suggestion
        // We don't generate builtin types, unless they are explicitly mapped
        // Because users need String.type (or others) to register their adapters
```

Reply:
Thinking a bit more about this. I think it makes sense to keep the builtin types in `apollo-api` always and not generate them. This means for builtin types that require a runtime adapter, users need to do something like:

```kotlin
apolloClientBuilder.addScalarAdapter(CompiledIDType, MyIDAdapter)
// instead of
// apolloClientBuilder.addScalarAdapter(ID.type, MyIDAdapter)
```

But I think this is such an advanced case that it's ok? It keeps the logic more symmetric. Builting types are in `apollo-api`, others get generated

Reply:
👍  I'll have a look!

Reply:
In retrospect, `CompiledIDType` should certainly have been `BuiltinIDType` but that'll be for v4 😅

Reply:
Hmm after more digging, the `CustomScalarAdapters` methods take instances of `CustomScalarType`s, whereas the compiled types are `ScalarType` - so `apolloClientBuilder.addScalarAdapter(ID.type, MyIDAdapter)` for instance doesn't work.

We could make `CustomScalarType` be a subclass of `ScalarType`, and change `CustomScalarAdapters` methods to take `ScalarType` instead. But that looks like a breaking change 😅

Or we can add `ScalarType` based methods (`responseAdapterFor`, `add`, `addCustomScalarAdapter`) in addition to the existing `CustomScalarType` based ones.

WDYT?

Reply:
Gosh this is hard. I'm not too much a fan of duplicating everything either. I think your initial approach of generating a `CustomScalarType` for builtin types if they have a mapping was best then. `CustomScalarType` will not be a super accurate name and the whole thing feels a bit asymmetrical but I don't see a better way.

Or should we always generate all builtin types? And mark `CompiledStringType`, etc.. as deprecated? I was initially not too fond of this because it puts `String`, `Int`, etc.. as top level classes but maybe since they're using a specific package name, that'll be ok...

Reply:
> Or should we always generate all builtin types?

I'll five it a go so we can see what it looks like :)